src/schema/sources.ts

```diff
@@ -983,6 +994,26 @@ export const resolvers: IResolvers<any, Context> = {
       await ensureSourcePermissions(ctx, id);
       return getSourceById(ctx, info, id);
     },
+    favoriteSources: async (
+      _,
+      { userId, limit }: FavoriteSourcesArgs,
+      ctx,
+      info,
+    ): Promise<GQLSource[]> => {
+      const sourceIds = await getFavoriteSourcesIds(ctx.con, userId, limit);
+
+      if (sourceIds.length === 0) {
+        return [];
+      }
+
+      return graphorm.query<GQLSource>(ctx, info, (builder) => {
+        builder.queryBuilder = builder.queryBuilder.andWhere(
+          `${builder.alias}.id IN (:...ids)`,
+          { ids: sourceIds },
+        );
+        return builder;
+      });
```

Comment:
Instead of doing that as two queries we can merge it into one.
We have the query builder so we can generate a query out of it and use it in the `IN` operator.
This is a suggestion, I understand the hackathon pressure